File: business/profile/src/main/java/cn/rongcloud/profile/dialog/UserInfoDialog.java
package cn.rongcloud.profile.dialog;

import android.content.Context;
import android.net.Uri;
import android.text.InputFilter;
import android.text.TextUtils;
import android.view.View;
import android.widget.EditText;
import android.widget.ImageView;

import com.basis.ui.BaseDialog;
import com.basis.utils.ImageLoader;
import com.basis.utils.KToast;
import com.basis.widget.ChineseLengthFilter;

import cn.rongcloud.config.UserManager;
import cn.rongcloud.profile.R;


public class UserInfoDialog extends BaseDialog implements View.OnClickListener {

    private OnUserListener listener;

    public interface OnUserListener {
        void onSave(String userName, Uri portrait);

        void onSelected();

//        void onLogout();
    }

    public UserInfoDialog(Context context, OnUserListener listener) {
        super(context, R.layout.layout_user_info_popup_window, true);
        this.listener = listener;
    }

    @Override
    public void initListener() {

    }

    private Uri selectedPicPath;
    private ImageView portrait;
    private EditText etName;

    @Override
    public void initView() {
        portrait = rootView.findViewById(R.id.iv_portrait);
        ImageLoader.loadUri(portrait, UserManager.get().getPortraitUri(), R.drawable.default_portrait);
        etName = rootView.findViewById(R.id.et_user_name);
        etName.setFilters(new InputFilter[]{new ChineseLengthFilter(20)});
        etName.setText(UserManager.get().getUserName());
        rootView.findViewById(R.id.iv_close).setOnClickListener(this);
        rootView.findViewById(R.id.tv_save_user_info).setOnClickListener(this);
        rootView.findViewById(R.id.tv_logout).setOnClickListener(this);
        portrait.setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        int id = v.getId();
        if (R.id.iv_close == id) {
            dismiss();
        } else if (R.id.tv_save_user_info == id) {
            String userName = etName.getText().toString().trim();
            if (TextUtils.isEmpty(userName)) {
                KToast.show(R.string.username_can_not_be_empty);
                return;
            }
            if (null != listener) listener.onSave(userName, selectedPicPath);
        } else if (R.id.tv_logout == id) {
//            if (null != listener) listener.onLogout();
        } else if (R.id.iv_portrait == id) {
            if (null != listener) listener.onSelected();
        }
    }


    public void setUserPortrait(Uri uri) {
        if (null == uri) return;
        selectedPicPath = uri;
        if (null != portrait) {
            ImageLoader.loadUri(portrait, selectedPicPath, R.drawable.default_portrait);
        }
    }
}

File: app/src/main/java/cn/rongcloud/voiceroomdemo/ui/dialog/UserInfoDialog.kt
/*
 * Copyright © 2021 RongCloud. All rights reserved.
 */

package cn.rongcloud.voiceroomdemo.ui.dialog

import android.content.Context
import android.net.Uri
import cn.rongcloud.voiceroomdemo.R
import cn.rongcloud.voiceroomdemo.common.*
import kotlinx.android.synthetic.main.layout_user_info_popup_window.*

/**
 * @author gusd
 * @Date 2021/06/08
 */
class UserInfoDialog(
    context: Context,
    private val logoutBlock: (() -> Unit)? = null,
    private val saveBlock: ((userName: String, portrait: Uri?) -> Unit)? = null,
    private val showPictureSelectBlock: (() -> Unit)? = null
) : BaseDialog(
    context, R.layout.layout_user_info_popup_window,
    false
) {

    private var selectedPicPath: Uri? = null

    override fun initListener() {

    }

    override fun initView() {
        iv_portrait.loadPortrait(AccountStore.getUserPortrait() ?: "")
        AccountStore.getUserName()?.let {
            et_user_name.setText(it)
        }
        iv_close.setOnClickListener {
            dismiss()
        }

        tv_save_user_info.setOnClickListener {
            if (et_user_name.text.isNullOrBlank()) {
                context.showToast(getString(R.string.username_can_not_be_empty))
                return@setOnClickListener
            }
            saveBlock?.invoke(et_user_name.text.toString(), selectedPicPath)
        }

        iv_portrait.setOnClickListener {
            showPictureSelectBlock?.invoke()
        }

        tv_logout.setOnClickListener {
            logoutBlock?.invoke()
        }

    }

    fun setUserPortrait(picturePath: Uri) {
        selectedPicPath = picturePath
        iv_portrait.loadLocalPortrait(picturePath)
    }
}